src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -84,14 +84,18 @@ protected override Size MeasureOverride(Size availableSize)
             {
                 _forceRemeasure = false;
                 _availableSpace = availableSize;
-                Controller?.UpdateControls();
             }
 
             return base.MeasureOverride(availableSize);
         }
 
         protected override Size ArrangeOverride(Size finalSize)
         {
+            if (_availableSpace != finalSize)
+            {
+                Controller?.UpdateControls();
```

Comment:
Btw, UpdateControls was already called in ArrangeOverride couple of lines below.